datafusion/physical-optimizer/src/enforce_sorting/sort_pushdown.rs

```diff
@@ -216,7 +218,28 @@ fn pushdown_sorts_helper(
 fn pushdown_requirement_to_children(
     plan: &Arc<dyn ExecutionPlan>,
     parent_required: OrderingRequirements,
+    parent_fetch: Option<usize>,
 ) -> Result<Option<Vec<Option<OrderingRequirements>>>> {
+    // Only attempt to push down TopK when there is an upstream LIMIT
+    if parent_fetch.is_some() {
```

Comment:
Can we pass down `parent.cardinality_effect()` instead? I would do `if !matches(parent.cardinality_effect(), CardinalityEffect::Equal)`

Reply:
Sorry @adriangb , i can't understand the parent here, what do you mean parent?

I can't find parent context from current sort pushdown code.

Reply:
I guess this code is already assuming that `parent` is some operator that has a limit. So yeah I agree with you not worth refactoring for this change, if there are still issues we can address them later.